src/graph_API.jl

```diff
@@ -0,0 +1,90 @@
+type edge
+  # param_node: the "in" node (listed first in connectparameter)
+  # var_node: the "out" node (listed second in connectparameter)
+  name::Symbol
+  param_node::Symbol #name of component
+  var_node::Symbol #name of component
+end
+
+type ModelGraph
+  edges
+  nodes
+  size
+end
+ModelGraph(e,n) = ModelGraph(e, n, length(n))
+ModelGraph() = ModelGraph([],[],0)
+
+function add_edge(g::ModelGraph, e::edge)
+  @assert e.param_node in g.nodes && e.var_node in g.nodes ["Tried to add an edge between vertices not in the graph"]
+  append!(g.edges, [e])
+end
+
+function add_node(g::ModelGraph, n::Symbol)
+  append!(g.nodes, [n])
+  g.size += 1
+end
+
+function get_all_edges(g::ModelGraph)
+  return g.edges
+end
+
+function get_all_nodes(g::ModelGraph)
+  return g.nodes
+end
+
+function print_graph(g::ModelGraph)
+  for c in get_all_nodes(g)
+    i_edges = get_edges(g,c,"incoming")
+    o_edges = get_edges(g,c,"outgoing")
+    println(c)
+    println("  incoming parameters:")
+    [println("    - ",e.name," from ",e.var_node) for e in i_edges]
+    println("  outgoing variables:")
+    [println("    - ",e.name," from ",e.param_node) for e in o_edges]
+  end
+end
+
+function string_representation(g::ModelGraph)
```

Comment:
The sole purpose of this function was just for me to be able to see if I was constructing the graph correctly; I just wrote it as an ad hoc way to loosely test functionality and (sort of) visualize connections between components. (this function will get deleted eventually)